airflow/models/taskinstance.py

```diff
@@ -2128,6 +2135,28 @@ def set_duration(self) -> None:
             self.duration = None
         self.log.debug("Task Duration set to %s", self.duration)
 
+    @provide_session
+    def _record_task_map_for_downstreams(self, value: Any, *, session: Session = NEW_SESSION) -> None:
```

Comment:
```suggestion
    def _record_task_map_for_downstreams(self, value: Any, *, session: Session) -> None:
```

I generally avoid using `provide_session` on internal funcs and favour being explicit about passing the session instead.